src/processes.c

```diff
@@ -2442,37 +2317,33 @@ static int ps_read (void)
 			pse.id       = procs[i].p_pid;
 			pse.age      = 0;
 
-			pse.num_proc = 1;
-			pse.num_lwp  = 1; /* XXX: accumulate p_tid values for a single p_pid ? */
-
-			pse.vmem_rss = procs[i].p_vm_rssize * pagesize;
-			pse.vmem_data = procs[i].p_vm_dsize * pagesize;
-			pse.vmem_code = procs[i].p_vm_tsize * pagesize;
-			pse.stack_size = procs[i].p_vm_ssize * pagesize;
-			pse.vmem_size = pse.stack_size + pse.vmem_code + pse.vmem_data;
-			pse.vmem_minflt = 0;
-			pse.vmem_minflt_counter = procs[i].p_uru_minflt;
-			pse.vmem_majflt = 0;
-			pse.vmem_majflt_counter = procs[i].p_uru_majflt;
-
-			pse.cpu_user = 0;
-			pse.cpu_system = 0;
-			pse.cpu_user_counter = procs[i].p_uutime_usec +
+			pse.gauges.num_proc = 1;
+			pse.gauges.num_lwp  = 1; /* XXX: accumulate p_tid values for a single p_pid ? */
+
+			pse.gauges.vmem_rss = procs[i].p_vm_rssize * pagesize;
+			pse.gauges.vmem_data = procs[i].p_vm_dsize * pagesize;
+			pse.gauges.vmem_code = procs[i].p_vm_tsize * pagesize;
+			pse.gauges.stack_size = procs[i].p_vm_ssize * pagesize;
+			pse.gauges.vmem_size = pse.gauges.stack_size + pse.gauges.vmem_code + pse.gauges.vmem_data;
+			pse.counters.vmem_minflt = procs[i].p_uru_minflt;
+			pse.counters.vmem_majflt = procs[i].p_uru_majflt;
+
+			pse.counters.cpu_user = procs[i].p_uutime_usec +
 						(1000000lu * procs[i].p_uutime_sec);
-			pse.cpu_system_counter = procs[i].p_ustime_usec +
+			pse.counters.cpu_system = procs[i].p_ustime_usec +
 						(1000000lu * procs[i].p_ustime_sec);
 
 			/* no I/O data */
-			pse.io_rchar = -1;
-			pse.io_wchar = -1;
-			pse.io_syscr = -1;
-			pse.io_syscw = -1;
-			pse.io_diskr = -1;
-			pse.io_diskw = -1;
+			pse.gauges.io_rchar = -1;
```

Comment:
I wonder if this can be accomplished by simply assigning `procstat_gauges_init` to `pse.gauges` before reading the values... In fact, can we do this right after we declare `pse`?